rs/state_manager/src/split.rs

```diff
@@ -199,9 +200,18 @@ fn write_checkpoint(
         MaliciousFlags::default(),
     );
 
+    let new_height = old_height.increment();
+
+    flush_page_maps(
```

Comment:
May be worth documenting why we're calling this.

Reply:
I added a comment, and also renamed the function to mention canister snapshots.